src/main/webapp/meal-page-script.js

```diff
@@ -19,15 +19,15 @@ function fetchMealInfo() {
   const urlParams = new URLSearchParams(queryString);
   const id = urlParams.get("id") ?? 0;
 
-  fetch("/meal/" + id.toString())
+  fetch(`/meal/${id.toString()}`)
```

Comment:
Nit: you don't need to call `toString()`. It'll be called implicitly in a string template.

`/meal/${id}` would do.